samples/lib/thrift/hello_client/src/main.cpp

```diff
@@ -24,11 +24,7 @@ using namespace apache::thrift;
 using namespace apache::thrift::protocol;
 using namespace apache::thrift::transport;
 
-#ifdef __ZEPHYR__
-void main(void)
-#else
 int main(int argc, char** argv)
-#endif
```

Comment:
Zephyr does not pass any arguments to the `main` function:

https://github.com/zephyrproject-rtos/zephyr/blob/b2532e3fb708a2b2e595819907720ea20e4989f4/kernel/init.c#L243

Since C++ allows the `main` function to be declared as `int main(void)`, it should be changed as follows:
```c
#ifdef __ZEPHYR__
int main(void)
#else
int main(int argc, char** argv)
#endif
```

p.s. we could potentially improve how the C++ main function is handled on the Zephyr side later.

Reply:
@stephanosio - Fixed!